lib/puppet/pops/lookup/interpolation.rb

```diff
@@ -90,14 +90,16 @@ def interpolate_method(method_key)
             lookup_invocation.report_found_in_overrides(root_key, ovr[root_key])
           else
             scope = lookup_invocation.scope
-            val = scope[root_key]
-            if val.nil? && !nil_in_scope?(scope, root_key)
-              defaults = lookup_invocation.default_values
-              if defaults.include?(root_key)
-                lookup_invocation.report_found_in_defaults(root_key, defaults[root_key])
-              else
-                nil
-              end
+            val = nil
+            if (default_val = lookup_invocation.default_values[root_key])
+              catch(:undefined_variable) { val = scope[root_key] }
+            else
+              val = scope[root_key]
+            end
+            if val.nil? && default_val.nil?
+              nil
+            elsif val.nil?
+              lookup_invocation.report_found_in_defaults(root_key, default_val)
```

Comment:
If someone sets `Puppet[:strict] = off` or `warning`, do we need to fallback to the old code, since `scope#[]` won't throw `:undefined_variable`?

Reply:
Ah I forgot the scope always throws, the behavior depends if the caller wraps the call with `catch` or not